From 23613542bc4ad90a419c876a89fdcae001d6c2c6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Cl=C3=A9ment=20DAVID?= <clement.david@3ds.com>
Date: Fri, 5 Jun 2026 15:10:32 +0000
Subject: [PATCH] C++20: fix implicit u8string to string conversion

This modification fixes the implicit conversion from u8string to string. Calling `std::path::string()` makes it system-dependent but typesafe.

The `test-types.cpp` file also contains an invalid wchar_t* to std::cout conversion.

See https://gitlab.com/scilab/scilab/-/merge_requests/1770

diff --git a/modules/ast/src/cpp/types/test-types.cpp b/modules/ast/src/cpp/types/test-types.cpp
index 7b0f36f1f..40b67c8fc 100644
--- a/modules/ast/src/cpp/types/test-types.cpp
+++ b/modules/ast/src/cpp/types/test-types.cpp
@@ -43,7 +43,7 @@ int main(void)
     ** types::String
     */
     types::String s42(L"42");
-    std::cout << "s42 = " << s42.get(0, 0) << std::endl;
+    std::wcout << L"s42 = " << s42.get(0, 0) << std::endl;
     assert(wcscmp(s42.get(0, 0), L"42") == 0);
     assert(s42.isInt() == false);
     assert(s42.isDouble() == false);
diff --git a/modules/core/src/cpp/inlinehelp.cpp b/modules/core/src/cpp/inlinehelp.cpp
index d2b10528e..41da5f0df 100644
--- a/modules/core/src/cpp/inlinehelp.cpp
+++ b/modules/core/src/cpp/inlinehelp.cpp
@@ -657,7 +657,7 @@ int generate_inline_links(const std::wstring& lang, const std::wstring& path)
                     }
                     else
                     {
-                        storedPath = replaceToolboxRoot(normalizedXmlPath, toolboxBase.u8string());
+                        storedPath = replaceToolboxRoot(normalizedXmlPath, toolboxBase.string());
                     }
                     links[id] = storedPath;
                 }
-- 
2.51.0
